packages/camera/lib/new/src/camera_controller.dart

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import 'common/camera_interface.dart';
+
+/// Controls a device camera.
+///
+/// Use [CameraController.availableCameras] to get a list of available cameras.
+///
+/// This class is used as a simple interface that works for Android and iOS.
+///
+/// When using iOS, simultaneously calling [start] on two [CameraController]s
```

Comment:
Thanks for clarifying!

Discussed offline, I'm thinking this is kind of a foot gun. Sounds like the current plugin behavior is to guard against this by disposing of multiple instances under the hood. I think we should probably maintain it here, but that doesn't need to block this PR.